syntax = "proto3";

package vectordb_meta;

message Replica {
  int32 id = 1;
  string name = 2;
  string table_name = 3;
  string partition_name = 4;
  int32 engine_type = 5;
  string address = 6;
  string path = 7;
}

message Partition {
  int32 id = 1;
  string name = 2;
  string table_name = 3;
  int32 replica_num = 4;
  int32 engine_type = 5;
  string path = 6;
  repeated Replica replicas = 7;
}

message Table {
  string name = 1;
  int32 partition_num = 2;
  int32 replica_num = 3;
  int32 engine_type = 4;
  string path = 5;
  repeated Partition partitions = 6;
}

message TableNames {
  repeated string table_names = 1;
}
